src/main/java/com/commercetools/sync/products/ProductSync.java

```diff
@@ -111,12 +110,18 @@ public ProductSync(@Nonnull final ProductSyncOptions productSyncOptions) {
     protected CompletionStage<ProductSyncStatistics> processBatch(@Nonnull final List<ProductDraft> batch) {
         productsToSync = new HashMap<>();
         draftsToCreate = new HashSet<>();
-        return productService.cacheKeysToIds()
+        existingDrafts = new HashSet<>();
```

Comment:
because in the `prepareDraftsForProcessing` method it adds into two of these collections. So unless I have a composite object having these collections, which could be returned by the method. Then it would be possible. I agree that it would be better for simpler unit testing the methods. But I would suggest it's separate PR then. https://github.com/commercetools/commercetools-sync-java/issues/239